fastparquet/dataframe.py

```diff
@@ -37,7 +38,7 @@ def empty(types, size, cats=None, cols=None, index_type=None, index_name=None):
     views = {}
 
     cols = cols if cols is not None else range(cols)
-    if isinstance(types, str):
+    if isinstance(types, str_type()):
```

Comment:
It would be nice not to have to call anything here.  I recommend doing the check at import time

```python
if PY2:
    str_type = ...
else:
    str_type = ...

from .util import str_type
```

Also we might want to create a separate compatibility.py files rather than util.  Or, as @pitrou suggests, simply use `six`.